model/src/main/scala/explore/data/Optics.scala

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2016-2020 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package explore.data
+
+import cats._
+import cats.implicits._
+import monocle._
+
+object Optics {
+  def filteredTraversal[T](predicate: T => Boolean): Traversal[List[T], T] =
```

Comment:
Should we propose some of these back to Monocle? Or put them in gsp-cors/gpp-ui

Reply:
I'm on the fence about this.

`zip` has been proposed in https://github.com/optics-dev/Monocle/issues/545, but it's unlawful if the `Lens`es overlap (hence the name `disjoint` here). But it does offer some convenience to have this instead of defining the zip manually each time we need it (it could be used here for example: https://github.com/gemini-hlsw/explore/blob/master/model/src/main/scala/explore/model/ModelOptics.scala#L44). I guess it could be added to Monocle's `unsafe` module, but I seem to remember that's going to be discarded in Monocle 3.

About `filteredTraversal`: TBH, we are not using it at the moment. I haven't checked if it's lawful.